### PR TITLE
Bugfix for issue #47 - idempotency issue while running on Docker container

### PR DESCRIPTION
**Pull request by cardil**
This bugfix should solve the issue #47.

It utilize the new (deprecated from start) wrapper fact of `$::jboss_virtual` which adds support for Docker even before Facter 2.1.x.

Code in `service.pp` manifest should use standard `$::virtual` fact after dropping support for Puppet 2.x

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/coi-gov-pl/puppet-jboss/pull/67" target="_blank">ReviewNinja

**Comment by cardil** [comment]
Review please @coi-gov-pl/jboss-puppet-team

File: lib/puppet_x/coi/jboss/facts.rb [hunk 1]
```diff
@@ -24,5 +24,17 @@ class << config
           end
         end
       end
+
+    # Check is dockerized
+    # Implementation is taken from Facter 2.1.x
+    # @deprecated TODO: remove afre droping support for Puppet 2.x
```
Comment (Cosaquee) [inline lib/puppet_x/coi/jboss/facts.rb]: typo :)

File: lib/puppet_x/coi/jboss/facts.rb [hunk 2]
```diff
@@ -24,5 +24,17 @@ class << config
           end
         end
       end
+
+    # Check is dockerized
```
Comment (Cosaquee) [inline lib/puppet_x/coi/jboss/facts.rb]: check if dockerized

File: lib/puppet_x/coi/jboss/facts.rb (hunk 1 above)
Comment (cardil) [inline lib/puppet_x/coi/jboss/facts.rb]: Type is being !fixed

**Pull request merged**